tensorflow/java/src/gen/cc/java_defs.h

```diff
@@ -0,0 +1,266 @@
+  }
+  static Type Wildcard() {
+    return Type(Type::GENERIC, "", "");
+  }
+  static Type ClassOf(const Type& type) {
+    return Class("Class").param(type);
+  }
+  static Type ListOf(const Type& type) {
+    return Interface("List", "java.util").param(type);
+  }
+  static Type IterableOf(const Type& type) {
+    return Interface("Iterable").param(type);
+  }
+  const Kind& kind() const { return kind_; }
+  const string& name() const { return name_; }
+  const string& package() const { return package_; }
+  const string& descr() const { return descr_; }
+  Type& descr(const string& descr) { descr_ = descr; return *this; }
+  const std::vector<Type>& params() const { return params_; }
+  Type& param(const Type& param) {
+    params_.push_back(param);
+    return *this;
+  }
+  const std::vector<Annotation>& annotations() const { return annotations_; }
+  Type& annotation(const Annotation& annotation) {
+    annotations_.push_back(annotation);
+    return *this;
+  }
+  const std::deque<Type>& supertypes() const { return supertypes_; }
+  Type& supertype(const Type& type) {
+    if (type.kind_ == CLASS) {
+      supertypes_.push_front(type);  // keep superclass at the front of the list
+    } else if (type.kind_ == INTERFACE) {
+      supertypes_.push_back(type);
+    }
+    return *this;
+  }
+  /// Returns true if "type" is of a known collection type (only a few for now)
+  bool IsCollection() const {
+    return name_ == "List" || name_ == "Iterable";
+  }
+  /// Returns true if this instance is a wildcard (<?>)
+  bool IsWildcard() const {
+    return kind_ == GENERIC && name_.empty();
+  }
+  bool operator==(const Type& type) const {
+      return name_ == type.name_ && package_ == type.package_;
+  }
+  bool operator!=(const Type& type) const { return !(*this == type); }
+
+ protected:
+  Type(Kind kind, const string& name, const string& package)
+    : kind_(kind), name_(name), package_(package) {}
+
+ private:
+  Kind kind_;
+  string name_;
+  string package_;
+  string descr_;
+  std::vector<Type> params_;
+  std::vector<Annotation> annotations_;
+  std::deque<Type> supertypes_;
+};
+
+/// \brief Definition of a Java annotation
+///
+/// This class only defines the usage of an annotation in a specific context,
+/// giving optionally a set of attributes to initialize.
+class Annotation : public Type {
+ public:
+  static Annotation Of(const string& type_name, const string& package = "") {
+    return Annotation(type_name, package);
+  }
+  const string& attrs() const { return attrs_; }
+  Annotation& attrs(const string& attrs) { attrs_ = attrs; return *this; }
+
+ private:
+  string attrs_;
+
+  Annotation(const string& name, const string& package)
+    : Type(Kind::ANNOTATION, name, package) {}
+};
+
+/// \brief A definition of a Java variable
+///
+/// This class defines an instance of a type, which could be documented.
+class Variable {
+ public:
+  static Variable Of(const string& name, const Type& type) {
```

Comment:
The "Of" here was a little confusing to me. This creates a "variable of type `type` named `name`", right?
Perhaps call it `Variable::Create(const string& name, const Type& type)`?

Reply:
Sorry, that was a terrible name, I was trying to avoid `Create` because I didn't want to prefix all my other factory methods with `Create` (i.e. `CreateIterableOf`, etc.) which makes it too verbose. But thinking of it now, I don't think I'll have to... so yes `Create` it will be :)